pysph/examples/solid_mech/oscillating_plate_gray.py

```diff
@@ -0,0 +1,250 @@
+                       's010', 's020', 's110', 's120', 's220', 'arho', 'au',
+                       'av', 'aw', 'ax', 'ay', 'az', 'ae', 'rho0', 'u0', 'v0',
+                       'w0', 'x0', 'y0', 'z0', 'e0')
+        plate.add_constant("amplitude_idx", amplitude_idx)
+
+        # get the minimum and maximum of the plate
+        xp_min = xp.min()
+        yp_min = yp.min()
+        yp_max = yp.max()
+        xw_upper, yw_upper = np.mgrid[
+            -self.plate_inside_wall_length:self.dx_plate / 2.:self.dx_plate,
+            yp_max + self.dx_plate:yp_max + self.dx_plate +
+            (self.wall_layers - 1) * self.dx_plate + self.dx_plate / 2.:
+            self.dx_plate]
+        xw_upper = xw_upper.ravel()
+        yw_upper = yw_upper.ravel()
+
+        xw_lower, yw_lower = np.mgrid[
+            -self.plate_inside_wall_length:self.dx_plate / 2.:self.dx_plate,
+            yp_min - self.dx_plate:yp_min - self.dx_plate -
+            (self.wall_layers - 1) * self.dx_plate - self.dx_plate / 2.:
+            -self.dx_plate]
+        xw_lower = xw_lower.ravel()
+        yw_lower = yw_lower.ravel()
+
+        xw_left_max = xp_min - self.dx_plate
+        xw_left_min = xw_left_max - (
+            self.wall_layers - 1) * self.dx_plate - self.dx_plate / 2.
+        yw_left_max = yw_upper.max() + self.dx_plate / 2.
+        yw_left_min = yw_lower.min()
+
+        xw_left, yw_left = np.mgrid[xw_left_max:xw_left_min:-self.dx_plate,
+                                    yw_left_min:yw_left_max:self.dx_plate]
+        xw_left = xw_left.ravel()
+        yw_left = yw_left.ravel()
+
+        # wall coordinates
+        xw, yw = np.concatenate((xw_lower, xw_upper, xw_left)), np.concatenate(
+            (yw_lower, yw_upper, yw_left))
+
+        # create the particle array
+        wall = get_particle_array(x=xw, y=yw, m=m, h=self.h,
+                                  rho=self.plate_rho0, name="wall")
+
+        add_properties(wall, 'cs', 'e', 'v00', 'v01', 'v02', 'v10', 'v11',
+                       'v12', 'v20', 'v21', 'v22', 'r00', 'r01', 'r02', 'r11',
+                       'r12', 'r22', 's00', 's01', 's02', 's11', 's12', 's22',
+                       'as00', 'as01', 'as02', 'as11', 'as12', 'as22', 's000',
+                       's010', 's020', 's110', 's120', 's220', 'arho', 'au',
+                       'av', 'aw', 'ax', 'ay', 'az', 'ae', 'rho0', 'u0', 'v0',
+                       'w0', 'x0', 'y0', 'z0', 'e0')
+        wall.set_lb_props(list(wall.properties.keys()))
+
+        return [plate, wall]
+
+    def create_solver(self):
+        kernel = CubicSpline(dim=2)
+        self.wdeltap = kernel.kernel(rij=self.dx_plate, h=self.h)
+
+        integrator = EPECIntegrator(plate=SolidMechStep())
+
+        dt = 1e-6
+        tf = 1
+
+        solver = Solver(kernel=kernel, dim=2, integrator=integrator, dt=dt,
+                        tf=tf, adaptive_timestep=False)
+        return solver
+
+    def create_equations(self):
```

Comment:
Would it be possible to make this into a simple scheme perhaps so we don't have to keep rewriting the equations?  Also can the example name be simply `oscillating_plate.py` so when more schemes are available, we can easily switch with the same example.